api/go.mod

```diff
@@ -15,67 +15,89 @@ require (
 require (
 	github.com/cert-manager/cert-manager v1.13.5
 	github.com/go-playground/validator/v10 v10.19.0
-	golang.org/x/exp v0.0.0-20240213143201-ec583247a57a
+	github.com/openstack-k8s-operators/openstack-operator/apis v0.0.0-20240418083003-41031f2a1340
```

Comment:
This will cause a dependency loop because we have the /api and the operator in the same repo and we use git commit based pseudoversions.

1. bump something in openstack-operator/api
2. dataplane-operator bumps to the new openstack-operator/api, this results in a new pseudoversion of dataplane-operator/api as well as they are in the same repo
3. now there is a need to bump openstack-operator to the new dataplane-operator/api version, but this will lead to a new pseudoversion of openstack-operator/api as well and you go to step 2 again

I afraid this will lead to pain in the future.


So if we want to keep the logic that the dataplane-operator reads the OpenStackVersion but want to break the cycle then we either need to
* stop using pseudoversions
* move the /api module to a separate repo
* move the openstack-operator and dataplane-operator into the same repo and use replace lines in the go.mod files to use the dependency from locally and not from github (like we do in all the main go.mod files towards the /api module already)

Reply:
That is a problem. I wonder if it would be possible to fashion some sort of quasi ControlPlane struct here and fill it with values obtained by client? I don't know how would that work, but it seems like the only way forward.

Reply:
We agreed on slack to allow the cycle to be added. We will disable automatic version bumping for this dependency in renovate and mid term we intend to merge the two operator repos together

Reply:
Ok. Let's do that then.